build fail: wasm_bpf on manylinux_2_28 (gcc 13)

### Summary

```
CMake Deprecation Warning at build/_deps/gtest-src/CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at build/_deps/gtest-src/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Downloading SIMDJSON source
-- Adding -Og to compile flag
-- Downloading SIMDJSON source -- done
-- Downloading the WASM spec test suite
-- Downloading the WASM spec test suite -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/core
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/core -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/multi-memory
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/multi-memory -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/tail-call
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/tail-call -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/extended-const
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/extended-const -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/threads
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/threads -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/function-references
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/function-references -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/gc
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/gc -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/exception-handling
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/exception-handling -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/exception-handling-legacy
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/exception-handling-legacy -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/relaxed-simd
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/relaxed-simd -- done
Downloading wasm-bpf
Downloaded wasm-bpf
-- Version: 10.2.1
-- Build type: Release
-- Build spdlog: 1.13.0
-- Build type: Release
-- Trying to get libbpf..
-- Build libbpf with pkg-config: OFF
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.4.2")
-- Try to get libbpf through PkgConfig
-- Checking for module 'libbpf>=1.2.0'
--   Package 'libbpf', required by 'virtual:world', not found
-- LIBBPF_FOUND=
-- libbpf not found using pkgconfig
-- Try to get libbpf through the pre-defined LIBBPF_SOURCE_DIR
-- LIBBPF_SOURCE_DIR not defined
-- Try to get libbpf through FetchContent
-- Fetching libbpf..
-- Fetched libbpf
-- SOURCE_ROOT=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src
-- LIBBPF_INCLUDE_DIRS=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/root/usr/include;/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/include/uapi;/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/include
-- LIBBPF_LIBRARIES=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/libbpf.so
-- LIBBPF_TARGET_NAME=libbpf
-- LIBBPF_LIBRARIES_STATIC=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/libbpf.a
-- LIBBPF_SOURCE=fetch-content
-- LIBBPF_DEP_LIBRARIES=elf;z
-- LIBBPF_DEP_LIBRARIES_STATIC=
-- Link libbpf statically
-- Configuring done (7.5s)
-- Generating done (0.1s)
-- Build files have been written to: /__w/WasmEdge/WasmEdge/build
[1/[29](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802302179?pr=3421#step:8:30)] Creating directories for 'libbpf'
[2/29] No download step for 'libbpf'
[3/29] No update step for 'libbpf'
[4/29] No patch step for 'libbpf'
[5/29] Performing configure step for 'libbpf'
[6/29] Generating /__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/simple_map.bpf.o, /__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/simple_ringbuf.bpf.o
make: Entering directory '/__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/bpf-sources'
rm -rf simple_map.bpf.o
clang -g -O2 -target bpf -D__TARGET_ARCH_x86 -c simple_map.bpf.c -o simple_map.bpf.o
llvm-strip -g simple_map.bpf.o
cp simple_map.bpf.o ../simple_map.bpf.o
rm -rf simple_ringbuf.bpf.o
clang -g -O2 -target bpf -D__TARGET_ARCH_x86 -c simple_ringbuf.bpf.c -o simple_ringbuf.bpf.o
llvm-strip -g simple_ringbuf.bpf.o
cp simple_ringbuf.bpf.o ../simple_ringbuf.bpf.o
make: Leaving directory '/__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/bpf-sources'
[7/29] Performing build step for 'libbpf'
FAILED: plugins/wasm_bpf/libbpf/src/libbpf-stamp/libbpf-build _deps/libbpf-src/src/build/libbpf.so _deps/libbpf-src/src/build/libbpf.a /__w/WasmEdge/WasmEdge/build/plugins/wasm_bpf/libbpf/src/libbpf-stamp/libbpf-build /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/libbpf.so /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/libbpf.a 
cd /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src && NO_PKG_CONFIG=1 OBJDIR=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build DESTDIR=/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/root CFLAGS=-fPIC make -C /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src install && /opt/_internal/pipx/venvs/cmake/lib/python3.10/site-packages/cmake/data/bin/cmake -E touch /__w/WasmEdge/WasmEdge/build/plugins/wasm_bpf/libbpf/src/libbpf-stamp/libbpf-build
make: Entering directory '/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src'
  MKDIR    /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/staticobjs
  CC       /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/staticobjs/bpf.o
  CC       /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/staticobjs/btf.o
btf.c:18:10: fatal error: gelf.h: No such file or directory
   18 | #include <gelf.h>
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:1[31](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802302179?pr=3421#step:8:32): /__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src/build/staticobjs/btf.o] Error 1
make: Leaving directory '/__w/WasmEdge/WasmEdge/build/_deps/libbpf-src/src'
[8/29] Generating /__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/bootstrap.bpf.o
make: Entering directory '/__w/WasmEdge/WasmEdge/build/_deps/wasm_bpf-src/examples/bootstrap'
clang -g -O2 -target bpf -D__TARGET_ARCH_x86 -I../../runtime/cpp/third_party/vmlinux/x86/ -I../../runtime/cpp/third_party/ -idirafter /opt/rh/gcc-toolset-13/root/usr/lib64/clang/17/include -idirafter /usr/local/include -idirafter /usr/include -c bootstrap.bpf.c -o bootstrap.bpf.o
llvm-strip -g bootstrap.bpf.o # strip useless DWARF info
make: Leaving directory '/__w/WasmEdge/WasmEdge/build/_deps/wasm_bpf-src/examples/bootstrap'
[9/29] Generating /__w/WasmEdge/WasmEdge/test/plugins/wasm_bpf/assets/runqlat.bpf.o
make: Entering directory '/__w/WasmEdge/WasmEdge/build/_deps/wasm_bpf-src/examples/runqlat'
clang -g -O2 -target bpf -D__TARGET_ARCH_x86 -I../../runtime/cpp/third_party/vmlinux/x86/ -I../../runtime/cpp/third_party/ -idirafter /opt/rh/gcc-toolset-13/root/usr/lib64/clang/17/include -idirafter /usr/local/include -idirafter /usr/include -c runqlat.bpf.c -o runqlat.bpf.o
runqlat.bpf.c:119:5: warning: __sync builtin operation MUST have natural alignment (consider using __atomic). [-Wsync-alignment]
  119 |     __sync_fetch_and_add(&histp->slots[slot], 1);
      |     ^
1 warning generated.
llvm-strip -g runqlat.bpf.o # strip useless DWARF info
make: Leaving directory '/__w/WasmEdge/WasmEdge/build/_deps/wasm_bpf-src/examples/runqlat'
[10/29] Building CXX object _deps/gtest-build/googletest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[11/29] Building CXX object _deps/gtest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
ninja: build stopped: subcommand failed.
```

### Appendix

[CI link](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802302179)

--- a/.github/actions/expand-variables/main.js
+++ b/.github/actions/expand-variables/main.js
@@ -12,6 +12,11 @@ const archList = [{
     docker_tag: 'manylinux2014_aarch64-plugins-deps',
     asset_tag: 'manylinux2014_aarch64',
 }, {
+    arch: x86_64,
+    runner: 'ubuntu-latest',
+    docker_tag: 'manylinux_2_28_x86_64-plugins-deps',
+    asset_tag: 'manylinux_2_28_x86_64',
+}, {
     arch: aarch64,
     runner: 'linux-arm64-v2',
     docker_tag: 'manylinux_2_28_aarch64-plugins-deps',

--- a/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
+++ b/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
@@ -2,7 +2,7 @@ ARG BASE=wasmedge/wasmedge:manylinux_2_28_x86_64
 FROM ${BASE}
 
 RUN cd && (yum check-update || true) && \
-  yum install -y wget unzip zlib-devel zlib-static elfutils
+  yum install -y wget unzip zlib-devel zlib-static elfutils-libelf-devel
 
 WORKDIR /root
 

